notebooks/05.log_and_register_fe_model.py

```diff
@@ -0,0 +1,243 @@
+# Databricks notebook source
+# MAGIC %pip install ./mlops_with_databricks-0.0.1-py3-none-any.whl
+
+# COMMAND ----------
+# dbutils.library.restartPython()
+
+# COMMAND ----------
+import mlflow
+from databricks import feature_engineering
+from databricks.feature_engineering import FeatureFunction, FeatureLookup
+from databricks.sdk import WorkspaceClient
+from mlflow.models import infer_signature
+from pyspark.sql import SparkSession
+from sklearn.compose import ColumnTransformer
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.impute import SimpleImputer
+from sklearn.metrics import accuracy_score, precision_score, recall_score, roc_auc_score
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import LabelEncoder, OneHotEncoder, StandardScaler
+
+from mlops_with_databricks.config import ProjectConfig
+
+# Initialize the Databricks session and clients
+spark = SparkSession.builder.getOrCreate()
+workspace = WorkspaceClient()
+fe = feature_engineering.FeatureEngineeringClient()
+
+# COMMAND ----------
+
+mlflow.set_registry_uri("databricks-uc")
+mlflow.set_tracking_uri("databricks")
+
+config = ProjectConfig.from_yaml(config_path="../project_config.yml")
+
+# Extract configuration details
+num_features = config.num_features
+cat_features = config.cat_features
+target = config.target
+parameters = config.parameters
+catalog_name = config.catalog_name
+schema_name = config.schema_name
+
+# Define table names and function name
+feature_table_name = f"{catalog_name}.{schema_name}.bank_marketing_features"
+
+# COMMAND ----------
+# Load training and test sets
+# Load training and test sets
+train_set = spark.table(f"{catalog_name}.{schema_name}.train_set")
+features_renamed = [f.replace(".", "_") if "." in f else f for f in train_set.columns]
+train_set = train_set.toDF(*features_renamed)
```

Comment:
this could be problematic for serving, I am afraid. the original train set must have the renamed features already - I have not tried this way, but if serving fails (can not find column names, and I am almost certain that would be the issue) this would be the reason.

Reply:
Thanks @mvechtomova  for taking a look. If you run this with "." in the column names it was failing, providing with very cryptic errors. 
Line 151 in my code was failing if I didn't replace "." in the columns, which is how the data came in from the source: 
`training_df = training_set.load_df().toPandas()`